### DISCUSSION TITLE
code example lines in epub file go outside margins + font size

### DISCUSSION THREAD

**Discussion by iexa** [sumatrapdfreader/sumatrapdf#3202] [General]
Hi,

This is an [epub book](https://anonfiles.com/pcrfo9P5y9/AW.Core.java.for.the.impatient.3rd.2022_epub), but most of them exhibit this problem. The layout is now mostly fine with epubs (since v3.3.3 when all spaces / tabs in code examples were collapsed into one) but sometimes when a line is long it is not wrapped and will go outside the margin(s).

[screenshot](https://i.ibb.co/k1K9qjK/Capture.jpg)

Also I I was not able to figure out a way to decrease the font size used in epubs. Is there a way to do that somehow?

**Comment by GitHubRulesOK** [comment]
I think those lines are intentionally not wrapped (wrapped with `<code>` or  `<pre>` preformatted text) thus the issue is the fixed display width is not enough for such long lines of code, a smaller font size in the source would solve that but SumatraPDF currently does not adjust content nor have style over-rides

I can manually edit the file to reduce font size to 9 where it says "small", but that is editing the source which seems to allow for cut off by providing above the code a link to image versions. However note if you copy the code it will include the unseen part.
![image](https://user-images.githubusercontent.com/12754913/210071998-b83be113-21d6-4031-b17a-23ec9555162c.png)

**Reply by iexa** [reply]
Thanks, I thought of editing the source css for the font-size but instead already tried selecting + copying and that is a good way to counteract / neutralize this problem.

I hope some kind of root font-size / base font-size possibility will be added sometime later to the app. The current zoom feature is not helping with this.

**Reply by GitHubRulesOK** [reply]
Cut and paste should not be an issue if includes line above and below. The image also shows the full width, but sadly all stuffed at the end of file.  The edit is very simple 
1) rename A COPY file to zip, (double click to view content)
2) copy out (extract) \OEBPS\9780138052027.css find `small` and edit I used 9pt as a guess
beware there are several the one you need for the shown case is
`.pre{font-family:Courier,Courier New,monospace;font-size:9pt;margin-top:.5pt;line-hei.....`
 but untested if that works for any longer cases
3)  then paste (overwrite) the edited file.css  over the original
4) rename zip to epub
